### PR TITLE
Fix EightPuzzle class implementation in search.py (#710)

### PR DESCRIPTION
**Pull request by SasankYadati**
In `search.py`, Environments / Problems are defined as classes which are derived from the `Problem` class. In these changes I made, I rewrote most of `EightPuzzle` class as a derived class of `Problem`. Ran the tests using py.test and also made sure the code is Python 3.4 complaint. Please review my changes and let me know your thoughts. Thanks.

### CODE REVIEW

**Comment by SasankYadati** [comment]
@MrDupin @norvig Could you please review my pull request? Thanks a lot.

**Comment by ad71** [comment]
A much required change! Also, remember to close #710 with your PR

File: search.py [hunk 1]
```diff
@@ -404,12 +404,84 @@ def astar_search(problem, h=None):
...
+        if goal:
+            self.goal = goal
+        else:
+            self.goal = [ [0,1,2], 
+                          [3,4,5], 
+                          [6,7,8] ]
+        Problem.__init__(self, initial, goal)
     
+    def find_blank_square(self, state):
+        """Return the index of the blank square in a given state."""
```
Comment (MrDupin) [inline search.py]: Since it's a one-line comment, the period at the end is not needed.

File: search.py [hunk 2]
```diff
@@ -404,12 +404,84 @@ def astar_search(problem, h=None):
...
+        for row in len(state):
+            for column in len(row):
+                if state[row][column]==0:
+                    index_blank_square = (row,column)
```
Comment (MrDupin) [inline search.py]: A space after the comma is needed.

**Comment by SasankYadati** [comment]
Very useful feedback. Thank you very much. I will address the styling issues and make a new PR.

**Pull request merged**